packages/zwave-js/src/lib/commandclass/BarrierOperatorCC.ts

```diff
@@ -132,6 +139,86 @@ export class BarrierOperatorCCAPI extends CCAPI {
 			await this.get();
 		}
 	}
+
+	public async getCapabilities(): Promise<readonly SignalType[]> {
+		this.assertSupportsCommand(
+			BarrierOperatorCommand,
+			BarrierOperatorCommand.CapabilitiesGet,
+		);
+
+		const cc = new BarrierOperatorCCCapabilitiesGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+		});
+		const response = (await this.driver.sendCommand<BarrierOperatorCCCapabilitiesReport>(
+			cc,
+			this.commandOptions,
+		))!;
+		return response.supportedSignalTypes;
+	}
+
+	// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+	public async getEvent(signalType: SignalType) {
+		this.assertSupportsCommand(
+			BarrierOperatorCommand,
+			BarrierOperatorCommand.EventGet,
+		);
+
+		const cc = new BarrierOperatorCCEventGet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			signalType,
+		});
+		const response = (await this.driver.sendCommand<BarrierOperatorCCEventReport>(
+			cc,
+			this.commandOptions,
+		))!;
+		return {
+			signalState: getEnumMemberName(SignalState, response.state),
+		};
+	}
+
+	public async setEvent(
+		signalType: SignalType,
+		signalState: SignalState,
+	): Promise<void> {
+		this.assertSupportsCommand(
+			BarrierOperatorCommand,
+			BarrierOperatorCommand.EventSet,
+		);
+
+		const cc = new BarrierOperatorCCEventSet(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			signalType,
+			signalState,
+		});
+
+		await this.driver.sendCommand(cc, this.commandOptions);
+
+		if (this.isSinglecast()) {
+			await this.getEvent(signalType);
+		}
+	}
+
+	protected [SET_VALUE]: SetValueImplementation = async (
+		{ property },
+		value,
+	): Promise<void> => {
+		if (typeof value !== "number") {
+			throwWrongValueType(this.ccId, property, "number", typeof value);
+		} else {
+			switch (property) {
+				case "state":
+				case "signalType":
+				case "signalState":
+					await this.set(value);
+					break;
+				default:
+					throwUnsupportedProperty(this.ccId, property);
+			}
```

Comment:
Yeah, that won't work. You have to use differnet commands for different properties. See the linked examples from the other comment.